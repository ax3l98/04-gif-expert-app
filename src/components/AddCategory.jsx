import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };
  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    setInputValue('');
    onNewCategory(inputValue.trim());
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gif"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
