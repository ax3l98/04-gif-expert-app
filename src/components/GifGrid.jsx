import { GifItem } from './';
import { useFetchGifs } from '../hooks';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando...</h2>}

      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
