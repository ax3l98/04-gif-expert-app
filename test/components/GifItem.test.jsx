import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem>', () => {
  const title = 'Naruto';
  const url = 'https://naruto-shippuden.com/naruto.jpg';

  test('should Debe de hacer match con el snapshot', () => {
    /* Tarea
      1. Añadir PropTypes ????? yarn add
        a. title obligatorio
        b. url obligatorio

      2. Evaluar el snapshot
    */

    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('should Debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug(); // Mostrar al 'sujeto de pruebas' en consola

    // expect(screen.getByRole('img').src).toBe(url); // Versión para componentes pequeños, aunque no se recomienda mucho su uso
    // expect(screen.getByRole('img').alt).toBe(title); // Versión para componentes pequeños, aunque no se recomienda mucho su uso

    const { src, alt } = screen.getByRole('img'); // Mejor utilizar esta manera
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should Debe de mostrar el título en el componente', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
