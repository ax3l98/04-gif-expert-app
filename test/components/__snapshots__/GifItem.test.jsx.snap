// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem> should Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://naruto-shippuden.com/naruto.jpg"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
